### PR TITLE
[M4] GitHub Actions

### LINKED ISSUE #47: [M4] Github action setup
Test runs before merging (maybe after marking PR ready for review), and automatic npm package publishing.

### PR DESCRIPTION
**Pull request by wertikalk**
Contains GH actions for: testing, auto publishing, and using the library

closes #47

### CODE REVIEW

File: .github/test-api-use/contracts/API-Inclusion.sol [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (snissn) [inline .github/test-api-use/contracts/API-Inclusion.sol:1]: Can we use the correct license here?

File: .github/workflows/publish-and-use.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                  node-version: "20"
+            - name: Install dependencies
+              run: |
+                  yarn install
+                  make install_solc_linux
+            - name: Compile contract file
+              run: make
+            - uses: JS-DevTools/npm-publish@v3
+              with:
+                  token: ${{ secrets.NPM_TOKEN }}
```
Comment (snissn) [inline .github/workflows/publish-and-use.yaml:24]: Does the repo have this saved as a secret or do we need to add it?

File: .github/workflows/security.yaml [hunk 3]
```diff
@@ -1,121 +1,86 @@
 name: "Security checks"
```
Comment (snissn) [inline .github/workflows/security.yaml:1]: Here is an example of the security check passing, correct? https://github.com/bojinovic/filecoin-solidity-library/actions/runs/9171124449

File: .github/workflows/publish-and-use.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+name: "Publish & use NPM package"
```
Comment (snissn) [inline .github/workflows/publish-and-use.yaml:1]: Here is an example of the publish and use action passing in a github action on the developers repo - https://github.com/bojinovic/filecoin-solidity-library/actions/runs/9175094751 - we need to confirm that this works on this repo after we merge the PR

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -1,977 +1,95 @@
-name: "Build contracts"
```
Comment (snissn) [inline .github/workflows/main.yaml:1]: Here looks like the main task and the related tests correctly running on the developers fork: https://github.com/bojinovic/filecoin-solidity-library/actions/runs/9063375701/job/24899183910

We need to confirm that this works on this repo after we merge it too

File: .github/test-api-use/contracts/API-Inclusion.sol [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        AccountAPI.authenticateMessage(target, params);
+    }
+
+    //AddressAPI
+    function actorid_conversion() public pure {
+        uint64 actorID = 1;
+        CommonTypes.FilAddress memory result = FilAddresses.fromActorID(actorID);
+    }
+    //DatacapAPI
+    function bundle(DataCapTypes.TransferParams memory params) internal returns (int256, DataCapTypes.TransferReturn memory) {
```
Comment (snissn) [inline .github/test-api-use/contracts/API-Inclusion.sol:35]: What is the expectation here related to code coverage? There are two DataCapAPI calls to name and symbol, but there are many other functions in the [DataCapAPI Library](https://github.com/filecoin-project/filecoin-solidity/blob/master/contracts/v0.8/DataCapAPI.sol). Do we need to have a full test here? Is the intention to show that the api is included and therefore a full comprehensive test is not needed here? Can the status be documented in a comment in this file?

**Review by snissn — CHANGES_REQUESTED** [review]
This looks appropriate to merge pending the resolution of a few questions and requests that I included in this review. Also I noted in the review that we want to validate that the gh actions run correctly on the target repo after the PR has been merged.

File: .github/test-api-use/contracts/API-Inclusion.sol (hunk 6 above)
Comment (wertikalk) [reply .github/test-api-use/contracts/API-Inclusion.sol:35]: This is a basic install and compile test after publishing new package version. It's just one contract that imports all *APIs and uses one function from each - meant to resemble the way lib user would interact with the lib.

`> Can the status be documented in a comment in this file?`
Not sure about 'status' means here?

Comment (snissn) [reply .github/test-api-use/contracts/API-Inclusion.sol:35]: No ok that makes sense - no need for any changes here - thank you for clarifying

**Review by VisionaireGlobal — APPROVED** [review]

**Pull request merged**